gdnative-core/src/core_types/quat.rs

```diff
@@ -0,0 +1,46 @@
+use crate::core_types::Real;
+use glam;
+
+#[derive(Copy, Clone, Debug, PartialEq)]
+#[repr(C)]
+pub struct Quat {
+    pub x: Real,
+    pub y: Real,
+    pub z: Real,
+    pub w: Real,
+}
+
+impl Quat {
+    #[inline]
+    pub fn new(x: Real, y: Real, z: Real, w: Real) -> Self {
+        Self { x, y, z, w }
+    }
+
+    // TODO rename
+    pub fn approx_eq(self, to: &Self) -> bool {
+        use super::ApproxEq;
+        self.x.approx_eq(&to.x)
+            && self.y.approx_eq(&to.y)
+            && self.z.approx_eq(&to.z)
+            && self.w.approx_eq(&to.w)
+    }
+
+    #[inline]
+    fn glam(self) -> glam::Quat {
+        self.into()
+    }
+}
+
+impl Into<glam::Quat> for Quat {
+    #[inline]
+    fn into(self) -> glam::Quat {
+        glam::Quat::from_xyzw(self.x, self.y, self.z, self.w)
+    }
+}
+
+impl From<glam::Quat> for Quat {
+    #[inline]
+    fn from(q: glam::Quat) -> Self {
+        Self::new(q.x, q.y, q.z, q.w)
+    }
+}
```

Comment:
We don't actually want these since we explicitly want to keep `glam` types out of the public API. They should be private or `pub(crate)` methods instead.